lib/ssh/src/ssh_connection.erl

```diff
@@ -739,21 +739,24 @@ handle_msg(#ssh_msg_channel_request{recipient_channel = ChannelId,
 handle_msg(#ssh_msg_channel_request{recipient_channel = ChannelId,
 				    request_type = "exit-signal",
 				    want_reply = false,
-				    data = Data},  
+				    data = Data},
            #connection{channel_cache = Cache} = Connection0, _, _SSH) ->
     <<?DEC_BIN(SigName, _SigLen),
-      ?BOOLEAN(_Core), 
+      ?BOOLEAN(_Core),
       ?DEC_BIN(Err, _ErrLen),
       ?DEC_BIN(Lang, _LangLen)>> = Data,
-    Channel = ssh_client_channel:cache_lookup(Cache, ChannelId),
-    RemoteId =  Channel#channel.remote_id,
-    {Reply, Connection} =  reply_msg(Channel, Connection0, 
-				     {exit_signal, ChannelId,
-				      binary_to_list(SigName),
-				      binary_to_list(Err),
-				      binary_to_list(Lang)}),
-    CloseMsg = channel_close_msg(RemoteId),
-    {[{connection_reply, CloseMsg}|Reply], Connection};
+    case ssh_client_channel:cache_lookup(Cache, ChannelId) of
+        #channel{remote_id = RemoteId} = Channel ->
+            {Reply, Connection} =  reply_msg(Channel, Connection0,
+                                             {exit_signal, ChannelId,
+                                              binary_to_list(SigName),
+                                              binary_to_list(Err),
+                                              binary_to_list(Lang)}),
+            CloseMsg = channel_close_msg(RemoteId),
+            {[{connection_reply, CloseMsg}|Reply], Connection};
+        _ ->
+            {[], Connection0}
```

Comment:
Maybe add a comment %% Connection already closed by peer

Reply:
you mean channel I guess. yes, good idea. I will.

Reply:
Yes I think you are correct about that it should be channel :)  I think I put to much emphasis on the {connection_reply, CloseMsg} part when I read it.  Maybe  
```
ChannelClose  =   channel_close_msg(RemoteId),
{[{connection_reply, ChannelClose}|Reply], Connection};
```
would help shift the attention to what is important? Although it is clear if you think a little longer about it!

Reply:
yes. applied.